pyproject.toml

```diff
@@ -48,6 +49,197 @@ Code = "https://github.com/SciTools/nc-time-axis"
 Discussions = "https://github.com/SciTools/nc-time-axis/discussions"
 Issues = "https://github.com/SciTools/nc-time-axis/issues"
 
+[tool.check-manifest]
+ignore = [
+  "src/nc_time_axis/_version.py",
+]
+
+[tool.codespell]
+ignore-words-list = "assertIn"
+skip = ".git,./docs/_build"
+
+[tool.mypy]
+disable_error_code = [
+    # TODO: exceptions that still need investigating are below.
+    # Might be fixable, or might become permanent (above):
+    "attr-defined",
+    "misc",
+    "no-untyped-call",
+    "no-untyped-def",
+    "unreachable",
+]
+enable_error_code = ["ignore-without-code", "redundant-expr", "truthy-bool"]
+strict = true
+warn_unreachable = true
```

Comment:
I'm guessing that the idea here is to signal that this is the intention while functionally turning this off with "unreachable" in the disabled error codes. I guess the alternative would be to have this be false but with a comment saying that's temporary. On reflection I think the approach here is probably best since it keeps all the TODOs in one place.

Reply:
I could comment it out and add a TODO, like I did with the `redundant-expr` rule:
```
enable_error_code = [
	"ignore-without-code",
#	"redundant-expr",		# TODO: Add back in when above ignores fixed
	"truthy-bool"
]
```

Reply:
If you think it would be neater that way, to be honest I could see an argument for doing it either way.

Reply:
I commented it out for the moment and added a comment to say it needs putting in when the error code is fixed; see bf0774f

This also required me to add an exception to repo-review (`MY103`)

It's probably better this way as previously I was sort of tricking repo-review that I had it set (then was sneakily putting in an ignore for it in mypy!)